src/cmd/list.go

```diff
@@ -179,33 +215,37 @@ func listImages() ([]map[string]interface{}, error) {
 		images = utils.SortJSON(images, "names", true)
 	}
 
-	return images, nil
+	var toolboxImages []toolboxImage
+	for _, image := range images {
+		var i toolboxImage
+		imageJSON, err := json.Marshal(image)
```

Comment:
I would have definitely expected the struct to be unmarshal'ed to when podman.GetImages() initially unmarshals the JSON. What's the reason for doing it here instead?

Reply:
Mainly, I wanted to keep the changes only in `src/cmd/list.go` and prevent any major refactors in other parts of the code that already rely on `podman.GetImages()`/`podman.GetContainers`.

Reply:
Unless I change `podman.GetImages()`/`podman.GetContainers()`, the additional marshaling needs to take place before unmarshaling into the struct.

Reply:
I understand not wanting to start changing more and more code, so perhaps this is an appropriate workaround for now. But I'd suggest a comment saying that this is a temporary workaround, and the unmarshalling and compatibility code should move to utils/podman.go - after all, having

```
                if podman.CheckVersion("2.0.0") {
                        idKey = "Id"
                } else {
                        idKey = "ID"
                }
```

in cmd/rm.go when it is using `GetContainers()` is not exactly where we want the code to end up either!

Reply:
I added a comment before both marshalling + unmarshalling sections.